dask/dataframe/shuffle.py

```diff
@@ -1073,7 +1056,7 @@ def compute_and_set_divisions(df, **kwargs):
 
 
 def set_sorted_index(df, index, drop=True, divisions=None, **kwargs):
-    if isinstance(index, _Frame):
+    if isinstance(index, Series):
```

Comment:
Why is this change here?

Reply:
If the type is a `DataFrame`, then it falls through to the next case, which is essentially the same, right? I guess, why not just raise the error right here since `MultiIndex` isn't supported? Anyhow, this change seems unrelated to the main point of the PR, so I don't see a reason to get caught up on it 😄

Reply:
This function is only called as a result of `DataFrame.set_index`. Since that is guaranteed to be passing previously through `check_index_column`, the error about `MultiIndex` should have been caught. As a matter of fact, now this is more clear with annotations: while in `core.set_index()` a `SequenceType[str]` is allowed, this is no longer the case in `shuffle.set_index()`, where it should be already a single column.

Reply:
The methods in `shuffle` are only called if all checks in `DataFrame.set_index()` have been susccessful. Among these checks, there's the one that checks that we don't try to index with a `DataFrame`. So at this point, `index` must be a `Series` or an `Index` instance. Checking for a `_Frame` is too general.

Annotations have been added to make it clear that a `Series` is expected.